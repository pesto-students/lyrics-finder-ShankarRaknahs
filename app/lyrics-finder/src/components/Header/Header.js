const Header = ({ firstName, lastName }) => {
  return (
    <section className='header'>
      <p className='firstname'> {firstName}</p>
      <p className='secondname'> {lastName}</p>
    </section>
  );
};

export default Header;
