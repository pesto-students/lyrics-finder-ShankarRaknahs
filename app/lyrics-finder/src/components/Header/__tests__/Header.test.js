import React from 'react';

import renderer from 'react-test-renderer';

// import { render } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import Header from '../Header';

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJSON from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });
const { mount } = Enzyme;

describe('<Header />', () => {
  it('render correctly (snapshot)', () => {
    const app = { firstName: 'Lyrics', secondName: 'Finder' };
    const tree = renderer.create(<Header {...app} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });

  it('render correctly using Enzyme (snapshot)', () => {
    const app = { firstName: 'Lyrics', secondName: 'Finder' };
    const tree = mount(<Header {...app} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
