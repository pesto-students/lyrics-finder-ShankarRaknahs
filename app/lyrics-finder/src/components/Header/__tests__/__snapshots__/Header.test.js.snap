// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> render correctly (snapshot) 1`] = `undefined`;

exports[`<Header /> render correctly using Enzyme (snapshot) 1`] = `
<Header
  firstName="Lyrics"
  secondName="Finder"
>
  <section
    className="header"
  >
    <p
      className="firstname"
    >
       
      Lyrics
    </p>
    <p
      className="secondname"
    >
       
      Finder
    </p>
  </section>
</Header>
`;
